File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        loop:
        for(int i=0; i < nums.length; i++){
            for(int j=0; j < nums.length; j++){
                if(nums[i]+nums[j]==target){
                    if(i==j){
                        continue;
                    }
                    result[0] = i;
                    result[1] = j;
                    break loop;
                }
            }
        }
        return result;
    }
}
// @lc code=end

File: 1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
    indexMap := make(map[int]int)
	for currIndex, currNum := range nums{
		if requiredIndex, isPresent := indexMap[target-currNum]; isPresent{
			return []int {requiredIndex, currIndex}
		}
		indexMap[currNum] = currIndex
	}
	return []int{}
}
// @lc code=end

